src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -87,8 +81,21 @@ public void InstallWorkloadPack(PackInfo packInfo, SdkFeatureBand sdkFeatureBand
                     {
                         if (!PackIsInstalled(packInfo))
                         {
-                            var packagePath = _nugetPackageDownloader.DownloadPackageAsync(new PackageId(packInfo.ResolvedPackageId), new NuGetVersion(packInfo.Version)).Result;
-                            tempFilesToDelete.Add(packagePath);
+                            string packagePath;
+                            if (string.IsNullOrWhiteSpace(offlineCache))
+                            {
+                                packagePath = _nugetPackageDownloader.DownloadPackageAsync(new PackageId(packInfo.ResolvedPackageId), new NuGetVersion(packInfo.Version)).Result;
+                                tempFilesToDelete.Add(packagePath);
+                            }
+                            else
+                            {
+                                _reporter.WriteLine(string.Format(LocalizableStrings.UsingCacheForPackInstall, packInfo.Id, packInfo.Version, offlineCache));
+                                packagePath = Path.Combine(offlineCache, $"{packInfo.ResolvedPackageId}.{packInfo.Version}.nupkg");
+                                if (!File.Exists(packagePath))
+                                {
+                                    throw new Exception(string.Format(LocalizableStrings.CacheMissingPackage, packInfo.ResolvedPackageId, packInfo.Version, offlineCache));
```

Comment:
If possible, it would probably be a lot better user experience if we provided the full list of packages that were missing from the offline cache, rather than just the first one we hit.